File: csharp/DesignPattern.Interpreter/VariableExpression.cs

namespace DesignPattern.Interpreter
{
    /// <summary>
    /// 变量，终结符表达式
    /// </summary>
    public class VariableExpression : Expression
    {
        private char key;
        public VariableExpression(char key)
        {
            this.key = key;
        }

        public override double Interpret(Context context)
        {
            return context.Variable[this.key];
        }
    }
}

File: spring-cloud-design-pattern/behavior-interpreter-pattern/src/main/java/org/springframework/cloud/pattern/expression/VariableExpression.java
package org.springframework.cloud.pattern.expression;

import org.springframework.cloud.pattern.context.Context;

/**
 * @author zhy
 * @date 2023/5/30 10:31
 */
public class VariableExpression implements AbstractExpression {
    private String name;

    public VariableExpression(String name) {
        this.name = name;
    }

    @Override
    public int interpreter(Context context) {
        return context.lookup(this);
    }

    @Override
    public String toString() {
        return name;
    }
}
